ibm/resource_ibm_satellite_cluster.go

```diff
@@ -0,0 +1,766 @@
+	_, err = waitForLocationNormal(location, d, meta)
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for getting location (%s) to be normal: %s", location, err)
+	}
+
+	if v, ok := d.GetOk("workers"); ok {
+		workerCount := int64(v.(int))
+		createClusterOptions.WorkerCount = &workerCount
+	}
+
+	if v, ok := d.GetOk("kube_version"); ok {
+		kube_version := v.(string)
+		createClusterOptions.KubeVersion = &kube_version
+	}
+
+	if res, ok := d.GetOk("zones"); ok {
+		zones := res.(*schema.Set).List()
+		for _, e := range zones {
+			r, _ := e.(map[string]interface{})
+			if ID := r["id"]; ID != nil {
+				zone := ID.(string)
+				createClusterOptions.Zone = &zone
+			}
+			break
+		}
+	}
+
+	if v, ok := d.GetOk("enable_config_admin"); ok {
+		enableConfigAdmin := v.(bool)
+		createClusterOptions.AdminAgentOptIn = &enableConfigAdmin
+	}
+
+	if v, ok := d.GetOk("pod_subnet"); ok {
+		podSubnet := v.(string)
+		createClusterOptions.PodSubnet = &podSubnet
+	}
+
+	if v, ok := d.GetOk("service_subnet"); ok {
+		serviceSubnet := v.(string)
+		createClusterOptions.ServiceSubnet = &serviceSubnet
+	}
+
+	if v, ok := d.GetOk("pull_secret"); ok {
+		pullSecret := v.(string)
+		createClusterOptions.PullSecret = &pullSecret
+	}
+
+	instance, response, err := satClient.CreateSatelliteCluster(createClusterOptions)
+	if err != nil {
+		return fmt.Errorf("Error Creating Satellite Location: %s\n%s", err, response)
+	}
+
+	d.SetId(*instance.ID)
+	log.Printf("[INFO] Created ROKS Satellite Cluster : %s", *instance.ID)
+
+	//Create workerpoolzone
+	workerPoolName := "default"
+	if res, ok := d.GetOk("zones"); ok {
+		zones := res.(*schema.Set).List()
+		if len(zones) >= 2 {
+			for i := 1; i < len(zones); i++ {
+				zone := zones[i].(map[string]interface{})
+				if ID := zone["id"]; ID != nil {
+					zoneId := ID.(string)
+					zoneOptions := &kubernetesserviceapiv1.CreateSatelliteWorkerPoolZoneOptions{
+						Cluster:    instance.ID,
+						Workerpool: &workerPoolName,
+						ID:         &zoneId,
+					}
+					if pathParamsMap != nil {
+						zoneOptions.Headers = pathParamsMap
+					}
+
+					response, err := satClient.CreateSatelliteWorkerPoolZone(zoneOptions)
+					if err != nil {
+						return fmt.Errorf("Error Adding Worker Pool Zone : %s\n%s", err, response)
+					}
+				}
+			}
+		}
+	}
+
+	if l, ok := d.GetOk("labels"); ok {
+		labels := make(map[string]string)
+		for k, v := range l.(map[string]interface{}) {
+			labels[k] = v.(string)
+		}
+
+		wpots := &kubernetesserviceapiv1.V2SetWorkerPoolLabelsOptions{
+			Cluster:    instance.ID,
+			Workerpool: &workerPoolName,
+			Labels:     labels,
+		}
+		if resourceGrp != "" {
+			wpots.XAuthResourceGroup = &resourceGrp
+		}
+
+		response, err := satClient.V2SetWorkerPoolLabels(wpots)
+		if err != nil {
+			return fmt.Errorf(
+				"Error updating the labels: %s\n%s", err, response)
+		}
+	}
+
+	//Wait for cluster to get warning state
+	_, err = waitForClusterToReady(*instance.ID, d, meta)
```

Comment:
Can we have wait_till parameter and wait for either one worker node ready or ingress ready similar to other clusters

Reply:
Fixed